### DISCUSSION TITLE
Certificate for signing desktop .air app

### DISCUSSION THREAD

**Discussion by NeronBurningRome** [airsdk/Adobe-Runtime-Support#3623] [General]
I'm trying to sign desktop app made in Flash Builder 4.7 IDE. Previously they used .p12 certificate. It was easy enough: while exporting .air file in the IDE just put link to certificate and password and it's signed. But what now? I tried to export .airi file and sign it with ssl eSigner CodeSignTool. But this file format is not supported. I've seen some workaround in other [post](https://github.com/airsdk/Adobe-Runtime-Support/discussions/3108) but they use digiCert tool and it's seems very complicated. 
Is there any simple method to sign air app?

**Comment by ajwfrost** [comment]
Support for simple certificate files and passwords is being withdrawn by the certificate authority companies, so there would need to be updated mechanisms... and there are of course multiple mechanisms because of the different vendors and the way they do things.

AIR signing is very similar to JAR signing, so if you have any instructions from your certificate provider on how to sign JAR files, you should be able to follow the same steps and provide the same arguments to ADT as you would to the jarsigner app.

I don't think there will be a "one size fits all" option here though because different certificate vendors will want different set-ups and they all need some kind of configuration. Looking at the SSL eSigner thing, it sounds like it's a web service that will send a hash of the files to the server to be signed - it may be possible to rename the .airi file to .jar and see if it works, but I suspect there may need to be some manual mechanism created to use their API for signing the files and package. We can look at the differences though, between .air file signing and .jar file signing, to see if we can actually make these similar enough that the tool would work also on .airi formats..

thanks

**Reply by NeronBurningRome** [reply]
Thanks for answer. It's surprising more people don't complain about it. It seems like a vital functionality. 
For me this whole signing process seems complicated. So if you don't know the answer I definitely won't find it. (I tried changing .airi extension to .jar, but it didn't help.)
I hope during holidays some developer will be bored with watching Die Hard and he will have time post some solution here:) 
Happy Holidays:)

**Reply by ajwfrost** [reply]
I know of a few using .air distribution still, but we recommend people switch to distributing native bundles that can be properly signed and packaged for the target operating systems. There are some restrictions to using the .air application installer, a key one being on macOS where applications should be code-signed and of course we can't do that step when we're creating and installing an application programmatically based on the .air contents. So then if you were using Windows, you'd just have to sign the generated executable file (and put it into an installer and sign the .msi too..) which is what we do with the AIR SDK Manager application. And then it's more "normal" for all the code-signing tools, and you'd be able to use the eSigner tools.

FYI - have got almost half way through Die Hard 2 with my daughter. We watched Die Hard itself last year. Classic Christmas movies :-)
